airflow/providers/amazon/aws/hooks/redshift_data.py

```diff
@@ -201,3 +216,54 @@ def get_table_primary_key(
                 break
 
         return pk_columns or None
+
+    async def check_query_is_finished_async(
```

Comment:
I have a general question about this function versus the `check_query_is_finished`. Why is their behaviour so different, if both are doing the same thing? The async version seems to include a loop that runs until the query stops running, whereas the sync version just does a check to see the current status. I think when we write functions like this (i.e. async versions of sync functions), we should try to keep the behaviour as similar as possible or rename the new function if it does something significantly different.

Reply:
yes, after reading it in detail, I think you're right. the loop stuff should be moved to trigger instead. thanks for your suggestion!